mitodl/ol-django
# Bump pluggy and faker versions, fix minor issue with base Dockerfile

### What are the relevant tickets?

n/a

### Description (What does it do?)

Generally:
- Update coverage to max 7.6.1
- Update pytest-cov to >4.0.0
- Swap pytest-lazyfixture with pytest-lazy-fixtures >=1

For `geoip`:
- Allow faker > 4.17 - newer projects (ecommerce) use newer versions of this, so hard-setting to this version breaks things

For `google_sheets_deferrals` and `google_sheets_refunds`:
- Allow pluggy > 0.13.1 - requiring 0.13.1 was causing a conflict trying to bootstrap this codebase

For `google_sheets`, `google_sheets_deferrals`, and `google_sheets_refunds`:
- Swap pytest-lazyfixture imports with pytest-lazy-fixtures ones

This also fixes a minor bug in the Dockerfile where the .cache folder is created with the wrong user. (There's a chown in there but mine kept being made as ubuntu:ubuntu - since we're doing stuff in ~dev it should probably just be as "dev".)

### How can this be tested?

Tests should pass. I did a quick scan of the changelogs for pluggy and faker and the APIs seemed stable so I don't expect issues.

It's easiest to run individual test suites by running `rye run pytest --no-cov tests/mitol/<app>`.
